### ISSUE TITLE
Workspace chat - Incorrect amount shown in preview card after payment.

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#46310]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.12-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Email or phone of affected tester (no customers):** biruknew45+2d2gg@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Prerequisite: A workspace is created and 2 expenses are submitted in the workspace chat.

1. Go to [staging.new.expensify.com].
2. Navigate to the workspace chat created.
3. Go to the expense details page.
4. Click on the first expense.
5. Click on the header, then select "hold."
6. Write any reason and then click on "hold."
7. Turn off the connection.
8. Click on the "pay with expensify" dropdown and then select "pay elsewhere."
9. Pay the amount.
10. Navigate back to the main workspace chat by clicking on the subtitle link.

## Expected Result:
After paying ETB 300, the preview card on the workspace chat page should show that ETB 300 is paid.

## Actual Result:
After paying ETB 300, the preview card on the workspace chat page incorrectly shows that ETB 400 is paid.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![Bug6553590_1721992163219!1](https://github.com/user-attachments/assets/8df64771-bdda-4051-8d7a-92e140b785ce)

https://github.com/user-attachments/assets/4ab664f7-f409-4794-84fc-145c3457f066

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
Production

https://github.com/user-attachments/assets/b5b0ecfc-73fe-4d6b-9d8b-5b9cd0070279

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After paying ETB 300, the preview card on the workspace chat page incorrectly shows that ETB 400 is paid.

### What is the root cause of that problem?
If we paid partial the expense report, we don't update the total of the expense report in optimistic data

https://github.com/Expensify/App/blob/afebc66349e21a5a7aa8ce6b04ab3c75e6e21620/src/libs/actions/IOU.ts#L6446

And then the `ReportPreview` show the amount is the old total amount
### What changes do you think we should make in order to solve the problem?
We should update the total of the expense report here if the paid partial

```
{
    onyxMethod: Onyx.METHOD.MERGE,
    key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport.reportID}`,
    value: {
        ...iouReport,
        lastMessageText: ReportActionsUtils.getReportActionText(optimisticIOUReportAction),
        lastMessageHtml: ReportActionsUtils.getReportActionHtml(optimisticIOUReportAction),
        hasOutstandingChildRequest: false,
        statusNum: CONST.REPORT.STATUS_NUM.REIMBURSED,
        pendingFields: {
            preview: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
            reimbursed: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
            partial: full ? null : CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
        },
        total: full ? iouReport.total : total
    },
},
```
https://github.com/Expensify/App/blob/afebc66349e21a5a7aa8ce6b04ab3c75e6e21620/src/libs/actions/IOU.ts#L6472
### What alternative solutions did you explore? (Optional)
NA

**Comment by nkdengineer** [comment]
@madmax330 Since it's a blocker, I can raise a PR quickly.

**Comment by francoisl** [comment]
Hi @nkdengineer, with any deploy blocker, can you please also link to the PR from the current checklist that caused the regression? Thanks!

**Comment by nkdengineer** [comment]
@francoisl It comes from this PR https://github.com/Expensify/App/pull/42573.

**Comment by kpadmanabhan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Preview card showing incorrect amount after payment while offline.

### What is the root cause of that problem?
When one payment was on hold, payment amount correctly reflects in "Pay XXX Elsewhere". But the same does not reflect in card amount. It still shows total amount before going offline.
Onyx is not updated while making part payments. Hence while offline, when we do not receive API response, we fetch data from onyx and display, which still contains the total amount. 

### What changes do you think we should make in order to solve the problem?
Update onyx on every payment irrespective of online or offline.

### What alternative solutions did you explore? (Optional)
N.A.

**Comment by nkdengineer** [comment]
### [Updated proposal](https://github.com/Expensify/App/issues/46310#issuecomment-2253085164)

- Add the fix for approving money request

**Comment by war-in** [comment]
Hi all 👋 I'm the author of the https://github.com/Expensify/App/pull/42573

We decided not to update the value of the old (original) report because we can have expenses in different currencies.

Example:

- report (A) has two expenses (A1, A2)
    - A1 - 10 USD
    - A2 - 30 some other currency which is 10x cheaper
- A2 has value = 13 USD
- we put A2 on hold
- we can't perform conversion on FE so we set A value to `13 - 30 = -17` which is of course wrong

That's why we can't do it and behaviour is like that

**Comment by luacmartins** [comment]
Ok, so this seems to be the expected behavior then. @war-in can you confirm so we can close this issue?

**Comment by war-in** [comment]
@luacmartins Yes, I confirm :)

**Comment by luacmartins** [comment]
Thanks! Gonna close this issue.
